fenix/app/src/test/java/org/mozilla/fenix/search/toolbar/ToolbarViewTest.kt

```diff
@@ -209,64 +210,145 @@ class ToolbarViewTest {
     }
 
     @Test
-    fun `GIVEN a general search engine is default WHEN a topic specific engine is selected THEN the hint changes`() {
-        val topicSpecificEngine = buildSearchEngine(SearchEngine.Type.BUNDLED, false)
+    fun `WHEN the default general search engine is selected THEN show text for default engine`() {
         val toolbarView = buildToolbarView(false)
-        toolbarView.update(defaultState)
+        val defaultEngine = buildSearchEngine(SearchEngine.Type.BUNDLED, true)
+
+        toolbarView.update(
+            defaultState.copy(
+                defaultEngine = defaultEngine,
+                searchEngineSource = SearchEngineSource.Default(defaultEngine),
+            ),
+        )
+
         assertEquals(context.getString(R.string.search_hint), toolbarView.view.edit.hint)
+    }
 
-        toolbarView.update(defaultState.copy(searchEngineSource = SearchEngineSource.Default(topicSpecificEngine)))
+    @Test
+    fun `WHEN a general search engine is selected THEN show text for general engine`() {
+        val toolbarView = buildToolbarView(false)
+        val generalEngine = buildSearchEngine(SearchEngine.Type.BUNDLED, true)
 
-        assertEquals(context.getString(R.string.application_search_hint), toolbarView.view.edit.hint)
+        toolbarView.update(
+            defaultState.copy(
+                searchEngineSource = SearchEngineSource.Shortcut(generalEngine),
+            ),
+        )
+
+        assertEquals(context.getString(R.string.search_hint_general_engine), toolbarView.view.edit.hint)
     }
 
     @Test
-    fun `GIVEN a topic specific engine is default WHEN a general engine is selected THEN the hint changes`() {
-        val topicSpecificEngine = buildSearchEngine(SearchEngine.Type.BUNDLED, false)
+    fun `WHEN a topic specific search engine is selected THEN show text for topic specific engine`() {
```

Comment:
I am not really happy that we are using the same string for both application and topic specific cases, but it aligns with [the design](https://www.figma.com/file/epu0Sj6mCPk7K282sNpHgI/Site?node-id=3754%3A51156&t=gYORIbyoTE2TYXJL-4). Maybe should create a follow up to separate those

Reply:
The `Enter search terms` string seems to be added for application search engines just as a safety net and so it would not really appear in the app for such engines?
Currently we have separate strings for each of the 3 application search engines.